app/dashboard/settings/profile/ProfileForm.tsx

```
@@ -0,0 +1,525 @@
+  }
+
+  const onSubmit = async (data: FormData, section: FormSection) => {
+    setIsSubmitting(true)
+    try {
+      // Create an object with only the modified fields
+      const modifiedData = Object.keys(dirtyFields).reduce((acc, key) => {
+        const value = data[key as keyof FormData]
+        // Handle null values for JSON fields
+        if (key === "badges") {
+          return {
+            ...acc,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)